Logging Addition

briancorbin opened the pull request.

briancorbin committed:
logging changes

kylefleming commented on Sources/Account/Account.swift:
```diff
@@ -229,7 +285,7 @@ extension OwnedTxOut {
             spentBlock = nil
         }
-
+        
```
Extra whitespace

briancorbin replied:
I checked all the whitespace, I think this might just be a visual thing with github

kylefleming replied:
It's definitely there, line 288, but there's other whitespace in the repo, so I can just remove it in a separate PR.

kylefleming replied:
Err, actually almost all the whitespace I'm noticing is being added by this PR. There's about 12 instances of trailing whitespace being added.

But it's a CBB so I wouldn't worry about it right now.

kylefleming commented on Sources/Encodings/MobUri.swift:
```diff
@@ -28,24 +33,29 @@ public enum MobUri {
     static func encode(_ payload: Payload) -> String {
-        "\(McConstants.MOB_URI_SCHEME)://\(payload.uriPath)"
+        logger.info("")
+        return "\(McConstants.MOB_URI_SCHEME)://\(payload.uriPath)"
```
`payload` is sensitive.

briancorbin replied:
payload isn't being logged here, just a blank log statement at the start of the encode function

kylefleming replied:
Oops! You're right

kylefleming commented on Sources/Fog/FogUntrustedTxOutFetcher.swift:
```diff
@@ -49,11 +54,13 @@ struct FogUntrustedTxOutFetcher {
                         "\(response)"))
                 }
-
+                
```
Extra trailing whitespace.

kylefleming commented on Sources/Fog/FogViewKeyScanner.swift:
```diff
@@ -20,16 +21,16 @@ final class FogViewKeyScanner {
-                    "containing \(blocksTxOuts.count) TxOuts")
+                logger.info(
+                    "\(blockRanges.map { $0.count }.reduce(0, +)) missed " +
+                        "blocks containing \(redacting: blocksTxOuts.count) TxOuts")
```
`blockTxOuts` isn't sensitive as it's the same for all users.

kylefleming commented on Sources/Fog/Report/FogResolver.swift:
```diff
@@ -23,6 +23,7 @@ final class FogResolver {
     }
 
     private init(attestation: Attestation) {
+        logger.info("attestation: \(attestation.description)")
```
Shouldn't need to add `.description` since it's `CustomStringConvertible`.

kylefleming commented on Sources/Fog/Report/FogResolverManager.swift:
```diff
@@ -17,6 +17,7 @@ final class FogResolverManager {
         serviceProvider: ServiceProvider,
         targetQueue: DispatchQueue?
     ) {
+        logger.info("fogReportAttestation: \(fogReportAttestation.description)")
```
Same here about `.description`.

kylefleming commented on Sources/Fog/View/FogRng.swift:
```diff
@@ -57,19 +58,22 @@ final class FogRng {
             }.mapError {
                 switch $0.errorCode {
                 case .invalidInput:
+                    logger.info("invalid key: \($0.description)")
```
LibMobileCoin errors are potentially sensitive.

kylefleming commented on Sources/Fog/View/FogRng.swift:
```diff
@@ -57,19 +58,22 @@ final class FogRng {
+                    logger.info("invalid key: \($0.description)")
                     return .invalidKey($0.description)
                 case .unsupportedCryptoBoxVersion:
+                    logger.info("unsupported crypto box version: \($0.description)")
```
Same here.

kylefleming commented on Sources/Fog/View/FogRng.swift:
```diff
@@ -117,6 +123,7 @@ final class FogRng {
     }
 
     func outputs(count: Int) -> [Data] {
+        logger.info("\(sensitive: count)")
```
`count` doesn't need to be sensitive.

kylefleming commented on Sources/Fog/View/FogRngSet.swift:
```diff
@@ -25,13 +25,20 @@ final class FogRngSet {
     ) -> FogSearchAttempt {
+        logger.info(
+            "requestedBlockCount: \(redacting: String(describing: requestedBlockCount)), " +
+                "numOutputs: \(redacting: numOutputs.value), " +
```
`numOutputs` doesn't need to be redacted.

kylefleming commented on Sources/Fog/View/FogRngSet.swift:
```diff
@@ -25,13 +25,20 @@ final class FogRngSet {
+        logger.info(
+            "requestedBlockCount: \(redacting: String(describing: requestedBlockCount)), " +
+                "numOutputs: \(redacting: numOutputs.value), " +
+                "minOutputsPerSelectedRnd: \(redacting: minOutputsPerSelectedRng)")
```
`minOutputsPerSelectedRng` doesn't need to be redacted. Also typo, should be `minOutputsPerSelectedRng`.

kylefleming commented on Sources/Fog/View/FogRngSet.swift:
```diff
@@ -52,6 +59,9 @@ final class FogRngSet {
         -> [Int64: RngTracker]
     {
+        logger.info(
+            "requestedBlockCount: \(redacting: String(describing: requestedBlockCount)), " +
```
`requestedBlockCount` doesn't need to be redacted.

kylefleming commented on Sources/Fog/View/FogRngSet.swift:
```diff
@@ -95,6 +105,8 @@ final class FogRngSet {
         highestProcessedBlockCount: UInt64
     ) -> Result<(), ConnectionError> {
+        logger.info("rngRecords: \(rngRecords), " +
+                "highestProcessedBlockCount: \(redacting: highestProcessedBlockCount)")
```
`highestProcessedBlockCount` doesn't need to be redacted.

kylefleming commented on Sources/Fog/View/FogRngSet.swift:
```diff
@@ -130,6 +143,8 @@ final class FogRngSet {
         highestProcessedBlockCount: UInt64
     ) -> Result<[FogView_TxOutSearchResult], ConnectionError> {
+        logger.info("txOutResults: \(redacting: txOutResults), " +
+                        "highestProcessedBlockCount: \(redacting: highestProcessedBlockCount)")
```
Same here about `highestProcessedBlockCount`.

kylefleming commented on Sources/Fog/View/FogRngSet.swift:
```diff
@@ -141,7 +156,9 @@ final class FogRngSet {
-                    assertionFailure("Error: \(Self.self).\(#function): RngTracker not found for " +
+                    logger.info("Error: RngTracker not found for rngKey in " +
+                                 "search attempt. ingestInvocationId: \(ingestInvocationId)")
+                    assertionFailure("Error: RngTracker not found for " +
```
This can be simplified to `logger.assertionFailure`.

kylefleming commented on Sources/Fog/View/FogRngSet.swift:
```diff
@@ -194,6 +211,7 @@ private final class RngTracker {
     var knownBlockCount: UInt64
 
     init(rng: FogRng, startBlockIndex: UInt64) {
+        logger.info("rng: \(rng), startBlockIndex: \(redacting: startBlockIndex)")
```
`startBlockIndex` doesn't need to be redacted.

kylefleming replied:
`FogRng` needs to be redacted. Also not sure it prints anything useful at the moment since it's a wrapper around a rust class.

kylefleming commented on Sources/Fog/View/FogRngSet.swift:
```diff
@@ -213,6 +232,8 @@ private final class RngTracker {
         highestProcessedBlockCount: UInt64
     ) -> Result<[FogView_TxOutSearchResult], ConnectionError> {
+        logger.info("rngSearchKeyToTxOutResult: \(redacting: rngSearchKeyToTxOutResult), " +
+                        "highestProcessedBlockCount: \(redacting: highestProcessedBlockCount)")
```
`highestProcessedBlockCount` doesn't need to be redacted.

kylefleming commented on Sources/Fog/View/FogRngSet.swift:
```diff
@@ -262,17 +289,22 @@ extension RngTracker {
     }
 
     convenience init(rng: FogRng, rngRecord: FogView_RngRecord) {
+        logger.info("rng: \(rng), rngRecord: \(rngRecord)")
```
Same here about `FogRng` being sensitive.

kylefleming commented on Sources/Fog/View/FogView.swift:
```diff
@@ -62,17 +65,23 @@ final class FogView {
-                    "error: \(error)")
+                logger.info(
+                    "Warning: could not decrypt TxOut returned from Fog View, base64 " +
+                        "ciphertext: \(searchResult.ciphertext.base64EncodedString()), " +
```
search result ciphertext is sensitive.

kylefleming commented on Sources/Fog/View/FogView.swift:
```diff
@@ -62,17 +65,23 @@ final class FogView {
+                logger.info(
+                    "Warning: could not decrypt TxOut returned from Fog View, base64 " +
+                        "ciphertext: \(searchResult.ciphertext.base64EncodedString()), " +
+                        "error: \(redacting: error)")
```
I believe `error` will have already been redacted, since it's not a LibMobileCoin error. If not we should make sure to redact it at the place where the error is created.

kylefleming commented on Sources/Fog/View/FogView.swift:
```diff
@@ -62,17 +65,23 @@ final class FogView {
             case .unsupportedVersion(let reason):
+                logger.info(
+                    "Warning: could not decrypt TxOut returned from Fog View, base64 " +
+                        "ciphertext: \(searchResult.ciphertext.base64EncodedString()), " +
```
Same here about redacting ciphertext.

kylefleming commented on Sources/Fog/View/FogView.swift:
```diff
@@ -62,17 +65,23 @@ final class FogView {
+                logger.info(
+                    "Warning: could not decrypt TxOut returned from Fog View, base64 " +
+                        "ciphertext: \(searchResult.ciphertext.base64EncodedString()), " +
+                        "error: \(redacting: error)")
```
And same here about not needing to redact non-LibMobileCoin errors.

kylefleming commented on Sources/Ledger/BlockMetadata.swift:
```diff
@@ -18,6 +18,7 @@ public struct BlockMetadata {
     }
 
     init(index: UInt64, timestamp: Date?) {
+        logger.info("index: \(index)")
```
We can log the timestamp.

kylefleming commented on Sources/Ledger/BlockMetadata.swift:
```diff
@@ -28,6 +29,7 @@ public struct BlockMetadata {
     }
 
     init(index: UInt64, timestampStatus: TimestampStatus?) {
+        logger.info("index: \(index)")
```
We can log the timestampStatus

kylefleming commented on Sources/Transaction/TransactionPreparer.swift:
```diff
@@ -41,10 +42,13 @@ struct TransactionPreparer {
                 sumOfValues: inputs.map { $0.value },
                 isGreaterThanValue: fee)
         else {
+            logger.debug("failure - sumOfValues inputs: \(redacting: inputs) " +
```
We should make this at least `info`, may `warn`?

briancorbin replied:
Agreed, I think warning is appropriate, I was gonna make all "failure" related logs warning level, if that sounds good?

kylefleming replied:
Sounds good to me.

briancorbin committed:
Addressing comments

Adjusted the log level of failure related logs to be "warning", and fixed the sensitivity level of some things to hide sensitive info and show non-sensitive info.

briancorbin committed:
Adding missed comment things

kylefleming reviewed (approved):
Thank you for this.

There are a couple things that'll need changing (like redacting to fee in 1 place, etc), but I'll handle those in a separate PR so that we can go ahead and get this landed.

🙏
